src/pages/play/Playground.tsx

```diff
@@ -0,0 +1,209 @@
+            updateModel(ref.current!.editor!.getModel()!);
+        }, 250),
+        [],
+    );
+
+    return (
+        <div className={styles.contentPane}>
+            <MonacoEditor
+                theme={theme}
+                language="typescript"
+                defaultValue={getInitialCode()}
+                options={commonMonacoOptions}
+                onChange={onChange}
+                ref={ref}
+            />
+        </div>
+    );
+}
+
+const astTheme = {
+    scheme: "monokai",
+    author: "wimer hazenberg (http://www.monokai.nl)",
+    base00: "#1e1e1e",
+    base01: "#383830",
+    base02: "#49483e",
+    base03: "#75715e",
+    base04: "#a59f85",
+    base05: "#f8f8f2",
+    base06: "#f5f4f1",
+    base07: "#f9f8f5",
+    base08: "#f92672",
+    base09: "#fd971f",
+    base0A: "#f4bf75",
+    base0B: "#a6e22e",
+    base0C: "#a1efe4",
+    base0D: "#66d9ef",
+    base0E: "#ae81ff",
+    base0F: "#cc6633",
+};
+
+const LuaSyntaxKind = __LUA_SYNTAX_KIND__;
+function LuaAST({ ast }: { ast: object }) {
+    const { isDarkTheme } = useThemeContext();
+    return (
+        <JSONTree
+            data={ast}
+            hideRoot={true}
+            theme={astTheme}
+            invertTheme={!isDarkTheme}
+            valueRenderer={(raw, value, lastKey) => {
+                if (lastKey === "kind") {
+                    return <em>{LuaSyntaxKind[value as any]}</em>;
+                }
+
+                return <em>{raw}</em>;
+            }}
+        />
+    );
+}
+
+function OutputPane() {
+    const theme = useMonacoTheme();
+    const { code, ast, results } = useContext(EditorContext);
+    const [isAstView, setAstView] = useState(false);
+    const toggleAstView = useCallback(() => setAstView(x => !x), []);
+
+    return (
+        <div className={styles.contentPane}>
+            <div className={styles.outputEditor}>
+                <div style={{ height: "100%", display: isAstView ? "none" : "block" }}>
+                    <MonacoEditor
+                        theme={theme}
+                        language="lua"
+                        value={code}
+                        options={{
+                            ...commonMonacoOptions,
+                            scrollBeyondLastLine: false,
+                            scrollBeyondLastColumn: 15,
+                            readOnly: true,
+                        }}
+                    />
+                </div>
+                <div style={{ height: "100%", overflow: "auto", display: isAstView ? "block" : "none" }}>
+                    <LuaAST ast={ast} />
+                </div>
+
+                <button
+                    className={clsx(
+                        "button button--outline button--primary",
+                        styles.outputToggleButton,
+                        !isAstView && "button--active",
+                    )}
+                    onClick={toggleAstView}
+                >
+                    {isAstView ? "AST" : "TEXT"}
```

Comment:
I had some trouble understanding which AST this was, can we rename this to "LUA AST"?